### PR TITLE
[Merged by Bors] - feat: port NumberTheory.ClassNumber.Finite

### PR DESCRIPTION
**Pull request by Genora51**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Genora51** [comment]
Currently struggling with a couple of errors that I can't seem to resolve, plus a timeout issue on `exists_mem_finsetApprox`.

**Comment by riccardobrasca** [comment]
Can you please fix the conflict please. Not sure I have time today, but I will have a look tomorrow.

**Comment by Genora51** [comment]
@riccardobrasca I've rebased onto latest `master` so should be all good to go.

File: Mathlib/NumberTheory/ClassNumber/Finite.lean [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+    exact fun h => hij ((distinctElems bS adm).injective h)
+#align class_group.mem_finset_approx ClassGroup.mem_finsetApprox
+
+section Real
+
+open Real
+
+attribute [-instance] Real.decidableEq
+
+set_option maxHeartbeats 800000 in
```
Comment (riccardobrasca) [inline Mathlib/NumberTheory/ClassNumber/Finite.lean:222]: This proof is rather slow. Maybe @Vierkantor wants to have a look at it (but it is already slow in mathlib3).

**Comment by riccardobrasca** [comment]
@Genora51 I think this is now green. I didn't fix any doc/name.

File: Mathlib/NumberTheory/ClassNumber/Finite.lean [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+        Ideal.dvd_iff_le.trans (by
+          rw [Equiv.refl_apply, Ideal.span_le, Set.singleton_subset_iff]; rfl)))
+    (ClassGroup.mkMMem bS adm) (ClassGroup.mkMMem_surjective L bS adm h)
+#align class_group.fintype_of_admissible_of_algebraic ClassGroup.fintypeOfAdmissibleOfAlgebraic
+
+/-- The main theorem: the class group of an integral closure `S` of `R` in a
+finite extension `L` of `K = Frac(R)` is finite if there is an admissible
+absolute value.
+
+See also `class_group.fintype_of_admissible_of_algebraic` where `L` is an
```
Comment (riccardobrasca) [inline Mathlib/NumberTheory/ClassNumber/Finite.lean:394]: This name is still the mathlib3 name.

**Review by jcommelin — COMMENTED** [review]
I suggest we look at speedups in follow-up PRs. The file looks good syntactically. Thanks :tada: 

bors merge

**Pull request closed without merging**